lib/livebook_web/components/app_components.ex

```diff
@@ -161,12 +164,6 @@ defmodule LivebookWeb.AppComponents do
 
   defp zta_options(), do: @zta_options
 
-  defp zta_metadata(nil), do: nil
```

Comment:
We need to handle nil here, so let's keep the private function and delegate to `Livebook.Config.zta_metadata` in the positive case.

Reply:
@jonatanklosko 

Do we need it? I'm not sure.

I was testing here. And when that form starts (initial state):

![CleanShot 2024-04-17 at 15 42 47](https://github.com/livebook-dev/livebook/assets/2719/d6dbbeef-484a-4e5b-b680-ff9263f49c2c)

The value inside `@form[:zta_provider].value` ([the argument that is passed](https://github.com/livebook-dev/livebook/blob/main/lib/livebook_web/components/app_components.ex#L135)) is `nil`.

And as we change the "Zero Trust Authentication provider" select, the value either changes to the provider type (an atom) or to `""` (if selected to "None").

Given this implementation:

```elixir
  def zta_metadata(zta_provider) do
    Enum.find(Livebook.Config.identity_providers(), &(&1.type == zta_provider))
  end
```

Passing `nil` or `""` to `Livebook.Config.zta_metadata/1` would be no problem; it just returns `nil`.

Actually, for any value that is not an existent provider name (as an atom), `Livebook.Config.zta_metadata/1` will return `nil`.

So, do you think we need to keep the private function? I'm not sure.

Reply:
> Actually, for any value that is not an existent provider name (as an atom), Livebook.Config.zta_metadata/1 will return nil.

This is an implementation detail that should not be known to the caller. The contract is that the function expects an atom type to be given, so I think it's better to be explicit and only ever call the function with a "valid" value :)

Reply:
Makes total sense. 👍 

Done in https://github.com/livebook-dev/livebook/pull/2572/commits/7a04940073da610237008a41be67669ee7e79794